src/components/app/createRoutes.jsx

```diff
@@ -14,9 +15,12 @@ export default function createRoutes(routes, authHoc = null) {
       );
     }
 
+    // Either standard or universal component
+    const component = route.eager ? route.component : universal(route.component, route.async || {});
+
     // handle public routes (no auth required, e.g: login)
     if (route.public || !authHoc) {
-      return <Route key={route.path} {...route} />;
+      return <Route key={route.path} {...Object.assign({}, route, {component: component})} />;
```

Comment:
@rsumilang can be simplified to `<Route key={route.path} {...route} component={component} />`